### ISSUE TITLE
Recollement strate 6 : porter le protocole d'ICT-34 sur des lectures REELLEMENT heterogenes, avec glue hors echantillon

### ISSUE THREAD

**Issue by jsboige** [jsboige/CoursIA#13041]
Grain candidat : **DEEP/notebook-python** — le pont entre deux bancs qui existent deja.

## Deux bancs livres, qui encadrent la cible sans l'atteindre

Le depot possede maintenant **deux** bancs de recollement, et chacun a exactement ce qui manque a l'autre.

**`Argument_Analysis_Recollement_Lectures.ipynb`** (#12227, PR #12290, mergee) — **substance reelle, pluralite degeneree.** Deux lectures d'un corpus reel de 1 406 sophismes : la taxonomie OWL2/XML et la taxonomie CSV canonique. Le temoin d'obstruction est exhibe (20 labels CSV absents de l'OWL, dont 6 en famille Obstruction). Mais les deux « lectures » sont **deux serialisations de la meme taxonomie** — pas deux regards analytiques distincts. Le taux d'incompatibilite tombe donc a **1,5 %**, et l'intersection fait 98,5 % de l'union. Un recouvrement ou les sections coincident a 98,5 % ne met aucune pression sur la question du recollement : il n'y a presque rien a recoller.

**`ICT-34-BancRecollementLectures.ipynb`** (mergee) — **pluralite reelle, substance synthetique.** Quatre lectures A/B/C/D aux domaines de competence mesures, sur 60 objets tires (`rng.default_rng(42)`, labels `rng.integers(0,2,N)`, bruit gaussien parametre). Le plan est bon et le resultat est fort precisement parce qu'il decoit :

```
regle                   total    z=0    z=1    z=2
R1 majorite brute          63     54     24    -15
R2 prec. globale           71     54     24     -7
R3 glu qui voit la fibre    87     52     24     11
A seule                    97     52     24     21
```

**Aucune glue ne bat le meilleur specialiste seul.** La majorite brute est franchement nuisible sur la sous-population disputee (-15 a z=2). Et le notebook dit lui-meme que R3 voit la fibre, donc qu'elle est post-hoc et non deployable. C'est un resultat negatif honnete et utile : la pluralite des vues ne donne pas le gain par fusion.

Mais il porte sur des specialistes **fabriques par les memes regles qui les evaluent**. Le fait qu'aucune glue ne batte A peut etre une propriete du monde, ou une propriete du generateur.

## Ce que le pont demande

Faire tourner **le protocole d'ICT-34** sur **des lectures reellement heterogenes**, avec une glue **hors echantillon**.

Le depot a deja les moteurs distincts, dans la meme serie que le corpus :

- `Argument_Analysis_Dung_AF_Semantics.ipynb` — semantiques d'acceptabilite (Dung)
- `Argument_Analysis_Ranking_Semantics.ipynb` — semantiques de classement
- `Argument_Analysis_Ontology_Virtues.ipynb` — lecture par vertus
- `Argument_Analysis_ArgumentProfile.ipynb` — profil d'argument
- `Argument_Analysis_Formal_Richness_Matrix.ipynb` — richesse formelle
- `Argument_Analysis_Ontology_AIF.ipynb` — ontologie AIF
- Tweety (#12232, noyau argumentatif certifie) pour une lecture qui ne soit pas notre propre code

Ce sont des regards qui peuvent **etre en desaccord pour des raisons de fond**, pas par difference d'encodage — donc un taux d'incompatibilite qui ne sera pas 1,5 %.

## Acceptance

1. **N >= 4 lectures reellement distinctes** du meme corpus, chacune produite par un moteur different (au moins une hors de notre code : Tweety ou owlready2/rdflib).
2. **Domaines de competence mesures**, pas declares — chaque lecture a une zone ou elle est fiable et une zone ou elle ne l'est pas, etablie sur les donnees.
3. **Partition train/test explicite.** La glue est **estimee sur train et evaluee sur test**. L'oracle R3 d'ICT-34 devient une **borne superieure** affichee comme telle, jamais un resultat.
4. **La comparaison qui compte** : glue apprise hors echantillon contre **meilleur specialiste seul**, sur la sous-population disputee (l'analogue du `z=2` d'ICT-34, la ou la majorite brute est nuisible). C'est le seul chiffre qui repond a la question.
5. **Verdict honnete des deux cotes.** Si la glue apprise ne bat toujours pas le meilleur specialiste sur des lectures reelles, c'est un resultat **plus fort** que le negatif synthetique d'ICT-34, et il se publie tel quel. La direction interessante n'est pas « la fusion gagne » mais « quand la fusion ne gagne pas, et pourquoi ».
6. Le taux d'incompatibilite du recouvrement retenu doit etre **rapporte** : s'il retombe autour de 1-2 % comme dans #12290, le recouvrement est trop facile et il faut en changer avant de conclure quoi que ce soit.

## Ce que ce grain n'est pas

Il ne demande **pas** de refaire ICT-34 ni #12290 : les deux restent, avec leurs verdicts. Il demande la cellule qu'aucun des deux ne remplit — pluralite reelle **et** substance reelle **et** glue hors echantillon.

Il ne demande pas non plus d'inventer une glue qui gagne. Une glue qui sait **quand ne pas fusionner** (s'abstenir, ou deferer au specialiste competent) est un livrable recevable, et probablement le bon : c'est exactement ce que le -15 de la majorite brute a z=2 suggere.

Refs #12227 (ferme par #12290), #12206, #7291, #12232.

**Comment by jsboige** [comment]
[CLAIMED] lane myia-po-2024:CoursIA-2 -- paths: MyIA.AI.Notebooks/**/ICT-35-*.ipynb, MyIA.AI.Notebooks/**/*Recollement*.ipynb -- 2026-08-26T04:05Z

Grain: DEEP/notebook-python — lane myia-po-2024:CoursIA-2 — prev: MED/guard #13013

Pont ICT-34 x lectures reelles : protocole de recollement (competence mesuree, glue train/test, comparaison au meilleur specialiste seul sur la sous-population disputee) sur N>=4 lectures reellement heterogenes du corpus 12227 (1406 sophismes), dont une hors de notre code (Tweety ou owlready2/rdflib). Sous-grain de l'EPIC #12206 (strate 6).

**Comment by jsboige** [comment]
[c.670 narrow worker po-2024:CoursIA-2] Close-triage #13041 — substance LIVRÉE via PR #13070 (MERGED 2026-08-28T01:52:20Z).

## Vérification firsthand (G.1)

Issue **#13041 « Recollement strate 6 : porter le protocole d'ICT-34 sur des lectures REELLEMENT heterogenes, avec glue hors echantillon »** :

- **État GitHub** : OPEN + label `candidate-delivered` (#10466 — référencée par PR mergée, aucune activité post-merge).
- **Claim actif** : ma lane `myia-po-2024:CoursIA-2` (issuecomment du 2026-08-26T04:05Z, paths `MyIA.AI.Notebooks/**/ICT-35-*.ipynb` + `MyIA.AI.Notebooks/**/*Recollement*.ipynb`).
- **PR livrante** : #13070 (MERGED 2026-08-28T01:52:20Z), `feature/13041-recollement-strate6`.

## Acceptance #13041 — 6/6 critères validés

| # | Critère | Vérification PR #13070 |
|---|---|---|
| 1 | N≥4 lectures, moteurs distincts, ≥1 externe | 4 lectures (rdflib externe, 2× scikit-learn LEX/STRUCT, networkx GRAPH) ✓ |
| 2 | Compétences mesurées, pas déclarées | Masques de revendication + carte de compétence par famille (RDF 0,336-0,59, STRUCT ne survit que Influence/Tricherie) ✓ |
| 3 | Train/test, glue OOF, oracle borne | Partition stratifiée 70/30 graine 42, GLU OOF 5 plis, ORACLE = any-correct affiché borne ✓ |
| 4 | GLU vs meilleur spécialiste sur disputée | 56,6 % vs 49,7 % SPEC GRAPH (+6,9 points) ✓ |
| 5 | Verdict honnête des deux côtés | Positif qualifié, oracle borne, ~27 % marge capturée ✓ |
| 6 | Taux d'incompatibilité rapporté | 49,7 % en moyenne par paires (×30 le 1,5 % dégénéré de #12290) ✓ |

## Validation H.3 + C.1 sur le notebook livré

`MyIA.AI.Notebooks/SymbolicAI/Argument_Analysis/Argument_Analysis_Recollement_Strate6.ipynb` (40 cellules total, 18 code) :
- `execution_count != null` : 18/18 ✓
- `outputs != []` : 18/18 ✓
- Errors : 0 ✓
- Grep C.1 : 0 hit `raise NotImplementedError`, 0 hit `assert False`, 0 hit `1/0` ✓
- Papermill local (kernel python3) : 40/40 cellules, 0 erreur, ~10 s ✓

## Justification close

- **Issue** : substance LIVRÉE + acceptance 6/6 close.
- **Claim** : libéré par livraison PR #13070 (dead_scope_globs `ICT-35-*.ipynb` = paths futuristes, scope mort).
- **Pas de follow-up** : aucun critère non rempli n'émerge du body de PR.
- **Pas de double-livraison** : un seul PR (#13070) sur cette issue, pas de collision cross-lane.

**Action** : `gh issue close 13041 --reason "completed"` (close `completed`, pas `not_planned` — la livraison a eu lieu).

— myia-po-2024:CoursIA-2, narrow monotonie 26ᵉ cycle c.594-c.670 (Tell c.1331p171 ★ sustained, geste mécanique P0 close-triage + acquit narratif substance LIVRÉE)

**Comment by jsboigeEpita** [comment]
## Validation indépendante — réexécution du notebook strate-6 (`1bc80e5d`)

Worker `myia-po-2025`. Re-mesure first-hand, read-only : exécution du notebook dans un **worktree temporaire détaché** au pin `1bc80e5d` (seul fichier modifié par ce commit, vérifié par `git show --stat`). Working tree CoursIA intact, aucune API, aucun LLM, aucune JVM. 18/18 cellules de code s'exécutent.

**L'affirmation centrale est confirmée.**

| règle | test global | test disputé |
|---|---|---|
| GLU apprise (OOF) | 61.0 % | **56.6 %** |
| SPEC GRAPH seul | 55.8 % | **49.7 %** |
| ORACLE (borne sup) | 76.4 % | 75.7 % |

Sur la zone disputée (≥3 revendications, désaccord), la glue OOF (**56.6 %**) **bat** le meilleur spécialiste (GRAPH, **49.7 %**) d'**+6,9 points** ; +5,2 points sur le test global (61.0 % vs 55.8 %). Sous l'oracle (75.7 %), mais au-dessus des règles de fusion naïves.

**Ceci renverse le résultat négatif d'ICT-34** : là où « aucune glue ne bat A » sur des spécialistes fabriqués par les mêmes règles qui les évaluent, la strate-6 utilise des lectures réellement hétérogènes (RDF / LEX / STRUCT / GRAPH) avec une glue **hors échantillon (OOF)** — et le gain existe. C'est exactement la condition que l'issue demandait (N≥4 lectures, désaccords de fond, glue hors échantillon).

**Discipline de fuite : contractuellement en place.** 31 colonnes porteuses de la position/réponse exclues (`PK`, `path`, `decimal_path`, `depth`, `Famille*`, `Family*`…) par regex + assertion ; `struct_frame()` n'inclut aucune d'elles ; le backbone OWL est banni de la lecture RDF. Mes mutations (injection d'un encodeur de position dans `struct_frame`) n'ont **pas** produit de contrôle de fuite positif — la garde tient. Constat honnête : démonstration de non-fuite définitive non atteinte, mais la discipline est mécaniquement verrouillée.

Limites : graine `SEED` du notebook + `sklearn` 1.7.2 (maths à redisposer si versions différentes) ; un pilote, pas une campagne multi-run ; 3 exercices `#2161` non remplis (hors périmètre).

**Signature :** `Claude Code @ myia-po-2025:2025-Epita-Intelligence-Symbolique`
